package/MDAnalysis/visualization/streamlines.py

```diff
@@ -149,30 +157,39 @@ def split_grid(grid, num_cores):
     # be distributed over the available cores on the system:
     list_square_vertex_arrays_per_core = np.array_split(list_all_squares_in_grid, num_cores)
     list_parent_index_values = np.array_split(list_parent_index_values, num_cores)
-    return [list_square_vertex_arrays_per_core, list_parent_index_values, current_row, current_column]
+    return (list_square_vertex_arrays_per_core, list_parent_index_values, current_row, current_column)
 
 
-def per_core_work(topology_file_path, trajectory_file_path, list_square_vertex_arrays_this_core, MDA_selection,
-                  start_frame, end_frame, reconstruction_index_list, maximum_delta_magnitude):
+def per_core_work(
+        topology_file_path: str, trajectory_file_path: str,
+        list_square_vertex_arrays_this_core: List[List[float]],
+        MDA_selection: str, start_frame: int, end_frame: int,
+        reconstruction_index_list: Sequence[int],
+        maximum_delta_magnitude: float,
+) -> List[Tuple[int, List[float]]]:
     """Run the analysis on one core.
 
     The code to perform on a given core given the list of square vertices assigned to it.
     """
     # obtain the relevant coordinates for particles of interest
     universe_object = MDAnalysis.Universe(topology_file_path, trajectory_file_path)
-    list_previous_frame_centroids = []
-    list_previous_frame_indices = []
+    list_previous_frame_centroids: List[int] = []
+    list_previous_frame_indices: List[int] = []
     #define some utility functions for trajectory iteration:
 
-    def produce_list_indices_point_in_polygon_this_frame(vertex_coord_list):
+    def produce_list_indices_point_in_polygon_this_frame(
+            vertex_coord_list: Iterable[int]) -> Iterable[int]:
         list_indices_point_in_polygon = []
         for square_vertices in vertex_coord_list:
             path_object = matplotlib.path.Path(square_vertices)
             index_list_in_polygon = np.where(path_object.contains_points(relevant_particle_coordinate_array_xy))
             list_indices_point_in_polygon.append(index_list_in_polygon)
         return list_indices_point_in_polygon
 
-    def produce_list_centroids_this_frame(list_indices_in_polygon):
+    # typing: numpy
+    def produce_list_centroids_this_frame(
+            list_indices_in_polygon: Iterable[int]
+    ) -> Union[List[np.ndarray], List[str]]:
         list_centroids_this_frame = []
```

Comment:
```
package/MDAnalysis/visualization/streamlines.py:224: error: Argument 1 to "__call__" of "ufunc" has incompatible type "Iterable[Any]"; expected "Union[Union[int, float, complex, str, bytes, generic], Sequence[Union[int, float, complex, str, bytes, generic]], Sequence[Sequence[Any]], _SupportsArray]"
```

I think the issue comes from here but I am not sure.

```suggestion
        list_centroids_this_frame: List[str, np.ndarray] = []
```